lib/logging/index.js

```diff
@@ -0,0 +1,576 @@
+  });
+};
+
+/**
+ * Get a reference to a Cloud Logging log.
+ *
+ * @resource [Log Overview]{@link https://cloud.google.com/logging/docs/api/ref_v2beta1/rest/v2beta1/projects.logs}
+ *
+ * @param {string} name - Name of the existing log.
+ * @return {module:logging/log}
+ *
+ * @example
+ * var log = logging.log('my-log');
+ */
+Logging.prototype.log = function(name) {
+  return new Log(this, name);
+};
+
+/**
+ * Get a reference to a Cloud Logging sink.
+ *
+ * @resource [Sink Overview]{@link https://cloud.google.com/logging/docs/api/ref_v2beta1/rest/v2beta1/projects.sinks}
+ *
+ * @param {string} name - Name of the existing sink.
+ * @return {module:logging/sink}
+ *
+ * @example
+ * var sink = logging.sink('my-sink');
+ */
+Logging.prototype.sink = function(name) {
+  return new Sink(this, name);
+};
+
+/**
+ * This method is called when creating a sink with a Bucket destination. The
+ * bucket must first grant proper ACL access to the Cloud Logging account.
+ *
+ * The parameters are the same as what {module:logging#createSink} accepts.
+ *
+ * @private
+ */
+Logging.prototype.setAclForBucket_ = function(name, config, callback) {
+  var self = this;
+  var bucket = config.destination;
+
+  bucket.acl.owners.addGroup('cloud-logs@google.com', function(err, apiResp) {
+    if (err) {
+      callback(err, null, apiResp);
+      return;
+    }
+
+    config.destination = 'storage.googleapis.com/' + bucket.name;
+
+    self.createSink(name, config, callback);
+  });
+};
+
+/**
+ * This method is called when creating a sink with a Dataset destination. The
+ * dataset must first grant proper ACL access to the Cloud Logging account.
+ *
+ * The parameters are the same as what {module:logging#createSink} accepts.
+ *
+ * @private
+ */
+Logging.prototype.setAclForDataset_ = function(name, config, callback) {
+  var self = this;
+  var dataset = config.destination;
+
+  dataset.getMetadata(function(err, metadata, apiResp) {
+    if (err) {
+      callback(err, null, apiResp);
+      return;
+    }
+
+    var access = [].slice.call(arrify(metadata.access));
+
+    access.push({
+      role: 'WRITER',
+      groupByEmail: 'cloud-logs@google.com'
+    });
+
+    dataset.setMetadata({
+      access: access
+    }, function(err, apiResp) {
+      if (err) {
+        callback(err, null, apiResp);
+        return;
+      }
+
+      config.destination = format('{baseUrl}/projects/{pId}/datasets/{dId}', {
+        baseUrl: 'bigquery.googleapis.com',
+        pId: dataset.parent.projectId,
+        dId: dataset.id
+      });
+
+      self.createSink(name, config, callback);
+    });
+  });
+};
+
+/**
+ * This method is called when creating a sink with a Topic destination. The
+ * topic must first grant proper ACL access to the Cloud Logging account.
+ *
+ * The parameters are the same as what {module:logging#createSink} accepts.
+ *
+ * @private
+ */
+Logging.prototype.setAclForTopic_ = function(name, config, callback) {
```

Comment:
I need to check if this will work, as it differs from our docs.
I'll get back to you.